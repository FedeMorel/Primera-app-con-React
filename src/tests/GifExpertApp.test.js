import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom'
import GifExpertApp from '../GifExpertApp';


describe('Pruevas en <GifExpertApp/>', ()=>{

    test('Debe demostarse correctamente el componente', ()=>{
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', ()=>{
        const categories = ['Batman', 'Hola'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })

})