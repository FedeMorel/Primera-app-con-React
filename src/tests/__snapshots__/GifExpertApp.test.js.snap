// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en <GifExpertApp/> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Batman"
      key="Batman"
    />
    <GifGrid
      category="Hola"
      key="Hola"
    />
  </ol>
</Fragment>
`;

exports[`Pruevas en <GifExpertApp/> Debe demostarse correctamente el componente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
